src/daemon/devnode.c

```diff
@@ -297,12 +327,12 @@ int rmdevpath(usbdevice* kb){
     euid_guard_start;
     int index = INDEX_OF(kb, keyboard);
     if(kb->infifo != 0){
-        int fd = kb->infifo - 1;
+        FILE* fd = fdopen(kb->infifo - 1, "a");
```

Comment:
This isn't actually an improvement on the original code; you go to the trouble of making a `FILE*` here, and then immediately use the old `int` fd in the `fcntl` call.

Reply:
I kind of wanted to use the FILE* and fputc approach because the code only requires one character to be written and doesn't require all the error handling that was necessary for the write system call.

To add to this, is it a bug in this case to open and then do fopen? Would that leave two dangling file descriptors?